examples/DynamicCantileverCase/dynamic_cantilever_visualization.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from matplotlib import pyplot as plt
+from elastica import *
+from dynamic_cantilever import simulate_dynamic_cantilever_with
+
+
+class DynamicCantileverSimulator(BaseSystemCollection, Constraints, CallBacks):
+    pass
+
+
+cantilever_sim = DynamicCantileverSimulator()
+
+# Options
+PLOT_FIGURE = True
+SAVE_FIGURE = False
+PLOT_VIDEO = True
+
+# Add test parameters
+n_elem = 200
+final_time = 100
+rendering_fps = 30
+mode = 2
+
+sim_result = simulate_dynamic_cantilever_with(
+    density=500,
+    n_elem=n_elem,
+    final_time=final_time,
+    mode=mode,
+    rendering_fps=rendering_fps,
+)
+
+cantilever = sim_result["rod"]
+recorded_history = sim_result["recorded_history"]
+omegas = sim_result["fft_frequencies"]
+amps = sim_result["fft_amplitudes"]
+vibration = sim_result["vibration"]
+peak = sim_result["peak"]
+
+# End position time-dependent plot
+fig = plt.figure(figsize=(20, 8), frameon=True, dpi=150)
+ax = fig.add_subplot(121)
+ax.plot(
+    recorded_history["time"],
+    recorded_history["z_position"],
+    lw=2.0,
+    label="Cosserat rod model",
+)
+ax.set_xlabel("Time [s]", fontsize=16)
+ax.set_ylabel("Displacement [m]", fontsize=16)
+ax.grid(visible=True, which="both", color="k", linestyle="--")
+
+times = np.array(recorded_history["time"])
+
+positions = vibration.get_positions(1, times)
+ax.plot(times, positions, lw=2.0, label="Euler-Bernoulli beam theory")
+ax.legend()
+
+# FFT plot
+fft_ax = fig.add_subplot(122)
+fft_ax.plot(omegas, amps, lw=2, color="k")
+fft_ax.set_xlim([0, vibration.get_omega() * 2])
+
+fft_ax.set_xlabel("Frequency [rad/s]", fontsize=16)
+fft_ax.set_ylabel("Amplitude [m]", fontsize=16)
+
+fft_ax.plot(omegas[peak], amps[peak], "*", color="red", ms=8)
+
+if PLOT_FIGURE:
+    plt.show()
+
+if SAVE_FIGURE:
+    fig.savefig("Dynamic_cantilever_visualization.png")
+
+if PLOT_VIDEO:
+    print("Plotting video ...")
+    video_name = f"Dynamic_cantilever_mode_{mode+1}.mp4"
+
+    import matplotlib.animation as manimation
+    from tqdm import tqdm
+
+    positions_over_time = np.array(recorded_history["position"])
+
+    FFMpegWriter = manimation.writers["ffmpeg"]
+    metadata = dict(title="Movie Test", artist="Matplotlib", comment="Movie support!")
+    writer = FFMpegWriter(fps=rendering_fps, metadata=metadata)
+    mvfig = plt.figure(figsize=(10, 8), frameon=True, dpi=150)
+    ax = mvfig.add_subplot(111)
+    ax.set_xlim([0, 1.25])
+    ax.set_ylim([-0.02, 0.02])
+    ax.set_xlabel("x [m]", fontsize=16)
+    ax.set_ylabel("z [m]", fontsize=16)
+    rod_lines_2d = ax.plot(
+        positions_over_time[0][0], positions_over_time[0][2], lw=8, color="k"
+    )[0]
+
+    with writer.saving(mvfig, video_name, dpi=150):
+        for t in tqdm(range(1, len(recorded_history["time"]))):
+            rod_lines_2d.set_xdata(positions_over_time[t][0])
+            rod_lines_2d.set_ydata(positions_over_time[t][2])
+            writer.grab_frame()
+
+    plt.close(plt.gcf())
```

Comment:
Lets move this to `post_processing.py`

Reply:
[70d0989](https://github.com/GazzolaLab/PyElastica/pull/173/commits/70d0989d02f60a73413ca195ca32d297702e7d34)